### DISCUSSION TITLE
Requesting a XML-Editable "signal_out" for modding

### DISCUSSION THREAD

**Discussion by BAB-Gaming** [FakeFishGames/Barotrauma#14450] [Ideas and suggestions]
Can we have it so there is an XML-Editable destination connection for the signal for when items have multiple components that all use the "signal_out" connection, below is an example (not sure if i defined the destconn_signal_out correctly):

```c#
item.SendSignal(signal, "signal_out");
```
replaced with:
```c#
[Serialize("signal_out", IsPropertySaveable.Yes, description: "The name of the connection the signal is sent to")]
public string destconn_signal_out  { get; set; }
----------------------------------------------------------------
item.SendSignal(signal, destconn_signal_out)
```
doing something like above will prevent this (hopefully):
```xml
<!-- <output name="state_out" displayname="connection.stateout" fallbackdisplayname="connection.signalout" />
 Removed due to detector signal confliction-->
```
(from my smart door)

**Comment by BlackMoons** [comment]
Agreed. Its been annoying that adding memory components to things ends up mixing signal out with other signal outs that are in the same component. Like mixing a controller and memory component results in the controller outputting on signal out, but so does the memory, making it rather useless.
